### ISSUE TITLE
[docs] torch.sigmoid to make clear equivalence relations to other sigmoid functions

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#105318]
### 📚 The doc issue

Currently https://pytorch.org/docs/stable/generated/torch.sigmoid.html only points to https://pytorch.org/docs/stable/special.html?highlight=expit#torch.special.expit

Both do not link to https://pytorch.org/docs/stable/generated/torch.nn.Sigmoid.html?highlight=sigmoid#torch.nn.Sigmoid or https://pytorch.org/docs/stable/generated/torch.Tensor.sigmoid.html?highlight=sigmoid#torch.Tensor.sigmoid

https://pytorch.org/docs/stable/special.html?highlight=expit#torch.special.expit (and other special functions) should deserve their own `.html` pages btw (as all other functions). expit should mention explicitly if it's just an alias to torch.sigmoid and torch.nn.functional.sigmoid

Ideally all these function variants should just say "See [some single variant]"

Currently there exists 4 functional variants and 1 module variant and findable in docs:
- torch.sigmoid
- torch.Tensor.sigmoid
- torch.nn.functional.sigmoid
- torch.special.expit
- torch.nn.Sigmoid

As usual, search results also contain duplicates and snippets are quite bad :(
<img width="305" alt="image" src="https://github.com/pytorch/pytorch/assets/1041752/adc1ff18-1ebf-45b4-be62-1ac6577d6b87">

**Comment by vadimkantorov** [comment]
Of course, instance variant / module variant / function variants are not exactly the same, but functionality is actually the same (and now some function variants suggest actually to see module variant for explanation and plot)

I don't mind if notation / plot is included in every variant, but still I propose it to have a note that the formula is equivalent to all other variants. I also don't mind if all variants link to a single variant (e.g. module variant as now it's where the plot is)

And I agree that it's a larger problem, but for sigmoid there is actually expit as well :)

Overall, I would propose to redirect all to one variant + maybe deprecate some variants: e.g. there should only be torch.sigmoid / tensor.sigmoid (and alias expit) and no more F.variant.

I think the users would benefit from actually deprecating older legacy variants and reducing the modern API surface. The same is very much needed for quantization and things like LinearReLU - it's almost impossible to understand all 9 variants :)

**Comment by mikaylagawarecki** [comment]
Actionable to make sigmoid variants point to a consistent place, as mentioned, perhaps `torch.nn.Sigmoid` as it contains the plot

**Comment by davidradl** [comment]
> Actionable to make sigmoid variants point to a consistent place, as mentioned, perhaps `torch.nn.Sigmoid` as it contains the plot

would you be ok with this going in as part of a glossary - it could be part of the starter set of words?

**Comment by davidradl** [comment]
@vadimkantorov sounds like we are looking for a similar goal. I raise issue [https://github.com/pytorch/pytorch/issues/104623](https://github.com/pytorch/pytorch/issues/104623). with the idea of having a conceptual overview, as part of that effort I think we need a glossary (not the contributor focussed GLOSSARY.md in the codebase). I think defining the concept of sigmoid in a new glossary then pointing to it from each API would be great. 

In terms of the deprecation, I do not know the process in the PyTorch for deprecating API, seems like removal would probably need to be on a new Version. I like that you are looking to rationalise the APIs, I am not sure of the impact, and documenting them all fully would be a non-contentions small step in the right direction.

**Comment by vadimkantorov** [comment]
IMO it's quite strange to put a concept of sigmoid in a glossary. IMO sigmoid function versions just need a slightly more consistent documentation

**Comment by davidradl** [comment]
> IMO it's quite strange to put a concept of sigmoid in a glossary. IMO sigmoid function versions just need a slightly more consistent documentation

Yes I can see that, I as thinking it would be a standard place we could put sigmoid information for PyTorch  (graph and concept) without it being tied to one API, I don't think any one API is more preferential to another. What do you think?

**Comment by vadimkantorov** [comment]
There are dozens of similar functions. I don't think glossary is the right place for this kind of things. But a document summarizing the PyTorch architecture and various subsystems is a valuable idea IMO. But I'm personally not a PyTorch maintainer, so my opinion isn't so important on this question.

**Comment by davidradl** [comment]
Sounds reasonable and other are likely to agree with you - lets put the extra details in one api then and point to it as you originally suggested.
